numpy/core/shape_base.py

```
@@ -187,22 +179,105 @@ def atleast_3d(*arys):
+        so for example a 2-D array of shape ``(M, N)`` becomes a view of
+        shape ``(M, N, 1, 1)`` when ``ndim=4``.
+
+    See Also
+    --------
+    atleast_1d, atleast_2d, atleast_3d
+
+    Notes
+    -----
+    This function does not follow the convention of the other
+    ``atleast_*d`` functions in numpy in that it only accepts a single
+    array argument. To process multiple arrays, use a comprehension or
+    loop around the function call. See examples below.
+
+    Setting ``pos=0`` is equivalent to how the array would be
+    interpreted by numpy's broadcasting rules. There is no need to call
+    this function for simple broadcasting. This is also roughly
+    (but not exactly) equivalent to
+    ``np.array(ary, copy=False, subok=True, ndmin=ndim)``.
+
+    It is easy to create functions for specific dimensions similar to
+    the other atleast_*d functions using Python's `functools.partial`
+    function. An example is shown below.
+
+    Examples
+    --------
+    >>> np.atleast_nd(3.0, 4)
+    array([[[[ 3.]]]])
+
+    >>> x = np.arange(3.0)
+    >>> np.atleast_nd(x, 2).shape
+    (1, 3)
+
+    >>> x = np.arange(12.0).reshape(4, 3)
+    >>> np.atleast_nd(x, 5).shape
+    (1, 1, 1, 4, 3)
+    >>> np.atleast_nd(x, 5).base is x.base
+    True
+
+    >>> [np.atleast_nd(x) for x in ((1, 2), [[1, 2]], [[[1, 2]]])]:
+    [array([[1, 2]]), array([[1, 2]]), array([[[1, 2]]])]
+
+    >>> np.atleast_nd((1, 2), 5, pos=0).shape
+    (1, 1, 1, 1, 2)
+    >>> np.atleast_nd((1, 2), 5, pos=-1).shape
+    (2, 1, 1, 1, 1)
+
+    >>> from functools import partial
+    >>> atleast_4d = partial(np.atleast_nd, ndim=4)
+    >>> atleast_4d([1, 2, 3])
+    [[[[1, 2, 3]]]]
+    """
+    ary = array(ary, copy=False, subok=True)
+    pos = normalize_axis_index(pos, ary.ndim + 1)
+    extra = ndim - ary.ndim
+    if extra > 0:
+        ind = pos * (slice(None),) + extra * (None,) + (Ellipsis,)
+        ary = ary[ind]
```

Two questions:
* What is supposed to happen if the type `ndmin` is invalid, _e.g._ if the user passes a `float`? 
* What is supposed to happen if `ndmin` is negative? Currently it just silently passes, which is actually consistent with `np.array(..., ndmin=...)` come to think of it.

---

I've added a check for anything that isn't a valid integer for `ndim`, and the appropriate tests. The behavior is slightly different than `np.array`, but I think you'll find it acceptable.

---

Negatives are passed though. I don't see any reason to do anything differently than `np.array` is doing.

---

> I don't see any reason to do anything differently than `np.array` is doing.

Yeah, I agree with your sentiment here.